### PR TITLE
Fix race when connection is canceled and new poll comes in

### PR DESCRIPTION
**Pull request by BrennanConroy**
#2700
There was a race where we could set the connection state to `Inactive` even though there was an active poll which could cause the connection to be closed by the server.

The fix was to add a "task queue" so all new poll requests would wait until the previous poll finished all state updates before starting the new poll state updates.

Details:
Browser could abort the http request (normal timeout behavior) and then send a new poll, on the server we might start processing the poll and see that there is an active poll which we could proceed to cancel and wait for the transport task to finish. The transport task would get an operation canceled exception and see that the http request was aborted so would just return normally, Then it would proceed to wait for the state lock which the new poll is currently holding, the new poll would see the transport task is finished and set the status as active and start the new transport task and then release the lock. Now the previous poll can aquire the lock, see that the connection is active (which it assumed was from when it polled) and set it to inactive. Then our heartbeat would come along 5 seconds later and clean the connection up.

TLDR:
Poll is canceled by client, client sends new poll, race between previous poll and new poll where status could be set as inactive and heartbeat could clean up connection.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs [hunk 1]
```diff
@@ -256,6 +257,8 @@ public HttpConnectionDispatcher(HttpConnectionManager manager, ILoggerFactory lo
                         // Start the transport
                         connection.TransportTask = longPolling.ProcessRequestAsync(context, tokenSource.Token);
 
+                        connection.PreviousPollTask = currentRequestTcs.Task;
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs]: We should move this outside the `if`, just in case a client re-polls during the first request. It would be rare because the first poll is "instant" but it could happen.

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs]: Good catch.

File: src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/LongPollingTransport.cs [hunk 2]
```diff
@@ -90,6 +90,8 @@ public LongPollingTransport(CancellationToken timeoutToken, PipeReader applicati
             catch (Exception ex)
             {
                 Log.LongPollingTerminated(_logger, ex);
+                context.Response.ContentType = "text/plain";
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/LongPollingTransport.cs]: This part of the change conflicts with https://github.com/aspnet/SignalR/commit/3d6e1e69fd03d0aed50f713ece482e7df65434a6. Revert it?

Comment (BrennanConroy) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/LongPollingTransport.cs]: Beat me to it, was just updating this
